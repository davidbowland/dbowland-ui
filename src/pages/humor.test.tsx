import React from 'react'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'

import Humor from './humor'

jest.mock('@fontsource/rokkitt')

jest.mock('@assets/jokes.yaml', () => ({
  jokes: ['Ha!', '=)'],
}))

describe('Humor page', () => {
  beforeAll(() => {
    const mockMath = Object.create(global.Math)
    mockMath.random = () => 0
    global.Math = mockMath
  })

  test('Ensure snapshot match', () => {
    const { container } = render(<Humor />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
