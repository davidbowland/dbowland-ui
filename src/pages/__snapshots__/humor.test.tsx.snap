// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Humor page Ensure snapshot match 1`] = `
<main>
  <title>
    Humor | dbowland.com
  </title>
  <h1>
    Ha!
  </h1>
  <button>
    Next joke
  </button>
</main>
`;
